### PR TITLE
Add a CONTRIBUTING.md file

### PR DESCRIPTION
**Pull request by timholy**
It seems this might help newcomers. I wasn't sure exactly where to host it, but this seems to be as good as any. We could also put it in the juliaimages docs, or at least put a link there to this repo.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+### Contributor Checklist
+
+* Create a [GitHub account](https://github.com/signup/free).
+
+* If you plan to fix a bug, feel free to first report the bug as an issue on its own.
+  In the text, you can mention whether you're planning on addressing it yourself.
+  *Pro tip*: if you do submit a pull request to fix it, put "Fixes #<issue number>" in the commit message and it will close automatically when your pull request is merged.
+
+  If you're concerned your change might be controversial, you can also use an issue to propose your change in general terms and discuss it before implementation.
+
```
Comment (johnnychen94) [inline CONTRIBUTING.md]: Perhaps add something like this?

"Not all codes are developed in Images.jl repository, if you don't know where to file an issue, try to identify the relevant package with `@which` macro."

Or add it to the `Before filing an issue` section?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+* Make your changes. Generally you should be working on a branch, so your work doesn't conflict with ongoing development in the `master` branch. Ensure you follow the [Julia style guide](https://docs.julialang.org/en/v1/manual/style-guide/index.html) for your contribution.
+
+* Test your changes. We aspire to have test coverage for every bit of "user visible" functionality. Tests are stored, appropriately, in the `test/` folder of each package. You can run existing tests yourself and add new ones. Sometimes testing is more work than the actual change itself, but having tests ensures that no well-meaning future developer will accidentally mess up your functionality---it's worth it!  *"A fix is for today. A test is forever."*
+
+* Submit your changes up to your fork and then submit a pull request---whoopee!
+
+* See what happens to the automated tests that run on Travis and/or AppVeyor. If there are errors, check the logs and see whether they look like they are related to your changes; if so, try to fix the problem by adding new commits to your pull request. Once the tests pass, hooray!
```
Comment (johnnychen94) [inline CONTRIBUTING.md]:
```suggestion
* See what happens to the automated tests that run on Travis and/or AppVeyor. If there are errors, check the logs and see whether they look like they are related to your changes; if so, try to fix the problem by adding new commits to your pull request. Once the tests pass, hooray! :tada:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+*By contributing you agree to be bound by JuliaImages' MIT license*
+
+If you've never submitted a pull request before, it can take a little while to become familiar with the process. In addition to the steps below, [GitHub has a tutorial and exercises](https://try.github.io/). See also the excellent [Git book](https://git-scm.com/book/en/v2). There are also many good external tutorials on this subject, like [this one](https://yangsu.github.io/pull-request-tutorial/).
+
+### Contributor Checklist
+
+* Create a [GitHub account](https://github.com/signup/free).
+
+* If you plan to fix a bug, feel free to first report the bug as an issue on its own.
```
Comment (johnnychen94) [inline CONTRIBUTING.md]:
```suggestion
* If you plan to fix a bug, feel free to first report the bug :bug: as an issue on its own.
```

File: README.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Images.jl
 
-An image processing library for [Julia](http://julialang.org/).
+An open-source image processing library for [Julia](http://julialang.org/).
```
Comment (johnnychen94) [inline README.md]:
```suggestion
An open-source image processing library for [Julia](http://julialang.org/). The core functionalities are developed in several packages under [JuliaImages](https://github.com/JuliaImages), [JuliaMath](https://github.com/JuliaMath) and [JuliaArrays](https://github.com/JuliaArrays).
```

File: README.md [hunk 5]
```diff
@@ -14,6 +14,13 @@ An image processing library for [Julia](http://julialang.org/).
 
 Full documentation is found at [JuliaImages](https://juliaimages.org/latest/).
```
Comment (johnnychen94) [inline README.md]:
```suggestion
Full documentation is found at [JuliaImages](https://juliaimages.org/stable/)
```
:P

**Comment by timholy** [comment]
Thanks for the review, @johnnychen94!

**Pull request merged**